### PR TITLE
Get CaltechHPC up and running again

### PR DESCRIPTION
**Pull request by knelli2**
## Proposed changes

- I edited the default number of cores to be 56 (with 2 charm nodes per physical node) so we can make use of our new reservation. If you decide to run on the 32 core nodes, you'll have to edit the submit script. Specifically, make the tasks per node only 1 now.
- Yaml-cpp told me to compile with `-fPIC` so I did.
- Load balancing doesn't work (as far as I know) with charm 7, so don't even bother including the Scotch LB in cmake
- Use system memory allocator so we can use pybindings.

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

**Review by nilsvu — APPROVED** [review]

**Comment by nilsvu** [comment]
> Use system memory allocator so we can use pybindings.

Note: pybindings should also work with jemalloc.

File: support/Environments/caltech_hpc_gcc.sh [hunk 1]
```diff
@@ -135,6 +135,7 @@ EOF
         cd build
         cmake -D CMAKE_BUILD_TYPE=Release -D YAML_CPP_BUILD_TESTS=OFF \
               -D CMAKE_C_COMPILER=gcc -D CMAKE_CXX_COMPILER=g++ \
+              -D CMAKE_CXX_FLAGS=-fPIC \
```
Comment (nilsvu) [inline support/Environments/caltech_hpc_gcc.sh]: The "right" way to do this is `-D CMAKE_POSITION_INDEPENDENT_CODE=ON`

File: support/Environments/caltech_hpc_gcc.sh [hunk 2]
```diff
@@ -295,7 +296,9 @@ EOF
 #%Module1.0
 setenv VIRTUAL_ENV $dep_dir/py_env
 prepend-path PATH ${VIRTUAL_ENV}/bin
+prepend-path PYTHONPATH ${VIRTUAL_ENV}/lib/python3.8/site-packages
```
Comment (nilsvu) [inline support/Environments/caltech_hpc_gcc.sh]: delete this line (shouldn't be needed, or did you encounter some problem?)

**Review by nilsvu — CHANGES_REQUESTED** [review]

File: support/Environments/caltech_hpc_gcc.sh [hunk 3]
```diff
@@ -356,10 +358,10 @@ spectre_run_cmake() {
     cmake -D CHARM_ROOT=$CHARM_ROOT \
           -D CMAKE_BUILD_TYPE=Release \
           -D CMAKE_Fortran_COMPILER=gfortran \
-          -D MEMORY_ALLOCATOR=JEMALLOC \
+          -D MEMORY_ALLOCATOR=SYSTEM \
           -D BUILD_PYTHON_BINDINGS=ON \
+          -D SPECTRE_TEST_RUNNER=mpiexec \
           -D Python_EXECUTABLE=`which python3` \
-          -D USE_SCOTCH_LB=ON \
           -D OVERRIDE_ARCH=skylake-avx512 \
```
Comment (nilsvu) [inline support/Environments/caltech_hpc_gcc.sh:365]: Add `-D MACHINE=CaltechHpc`

**Review by nilsvu — COMMENTED** [review]
I'd say drop the first couple of commits that I moved to #5007. I think we can just merge this and keep testing. We can run a simulation with this.

**Comment by knelli2** [comment]
Ok done!

**Review by nilsvu — DISMISSED** [review]

**Review by nilsvu — APPROVED** [review]

**Pull request merged**